src/redux/actions/index.js

```diff
@@ -34,3 +40,54 @@ export function setLanguageName(language) {
     payload: language,
   };
 }
+
+export function setLearntPhrases(phrase) {
+  return {
+    type: SET_LEARNT_PHRASES,
+    payload: phrase,
+  };
+}
+
+export function addLearntPhrases(phrase) {
+  return async dispatch => {
+    const storedLearntPhrases = await getDataFromStorage(LEARNT_PHRASES_KEY);
+    let itemsToStore = null;
+
+    if (!storedLearntPhrases) {
+      itemsToStore = [phrase];
+    } else {
+      itemsToStore = [...storedLearntPhrases, phrase];
+    }
+
+    await setDataToStorage(LEARNT_PHRASES_KEY, itemsToStore);
+
+    dispatch(setLearntPhrases(itemsToStore));
+    return Promise.resolve();
+  };
+}
+
+// Updating learntPhrases after moving the incorrect anwers to seen phrases
+export function removeWrongAswerFromLearntPhrases(phrase) {
+  return async dispatch => {
+    const storedLearntPhrases = await getDataFromStorage(LEARNT_PHRASES_KEY);
+    const learntPhrasesWithoutWrongAnswer = storedLearntPhrases.filter(
```

Comment:
please check if its working with empty storage you might need check if it exists: 

storedLearntPhrases?.filter